### DISCUSSION TITLE
Lesson4: Don't quite understand the contract code in the challenge

### DISCUSSION THREAD

**Discussion by fushengcheng** [Cyfrin/foundry-full-course-cu#605] [Q&A] [answered]
In line 40 of the contract code the actualPrice is multiplied by 10e10, but in a similar place in the video 1e10 is used. I don't quite understand why 10e10 is used here instead of 1e10. I personally understand that if we use 1e10 we can make actualPrice have 18 decimals like Ether.
Here's the code:
```
function solveChallenge(uint256 priceGuess, string memory yourTwitterHandle) external {
        uint256 actualPrice = getPrice();
        if (getDecimals() == 8) {
            actualPrice = actualPrice * 10e10; // ?
        }
        if (actualPrice + 3e18 > priceGuess && actualPrice - 3e18 < priceGuess) {
            _updateAndRewardSolver(yourTwitterHandle);
        } else {
            revert LessonFour__WrongPrice();
        }
    }
```
Here's the link:
[https://sepolia.etherscan.io/address/0x1b30DA2a868704483143A4D46865Ac9585629fd0#code](url)

**Comment by 0xusmanf** [comment, marked as answer]
Yes, you are right.
> actualPrice = actualPrice * 10e10; // ?

but calculations in the challenge contract are intentional so that you don't just look up the price on coinmarketcap and solve the challenge.

Patrick wants you to do your own calculations in the challenge.

**Reply by fushengcheng** [reply]
I get it. Thanks for the answer.
